plone/app/contenttypes/browser/link_redirect_view.py

```diff
@@ -111,9 +111,7 @@ def display_link(self):
             obj = uuidToObject(uid)
             if obj:
                 title = obj.Title()
-                meta = "/".join(obj.getPhysicalPath()[2:])
-                if not meta.startswith("/"):
-                    meta = "/" + meta
+                meta = "/" + obj.absolute_url_path()
```

Comment:
absolute_url_path should give a url in form of: /Plone/my-folder, thats okay.

Reply:
done in https://github.com/plone/plone.app.contenttypes/commit/21ff30d72e77dca9cd980fa03994fabfd25f508d

Reply:
test passes and locally using site urls or localhost or _vh_ works as expected. I've tested with a local vhost:
for `${navigation_root_url}/sitemap`

```
<VirtualHost *:80>
        ServerName test.com
[...]
        RewriteEngine on
        RewriteRule ^/sub(.*) http://127.0.0.1:8080/VirtualHostBase/http/test.com:80/Plone/VirtualHostRoot/_vh_sub$1 [L,P]
```
returns `/sub/sitemap` for portal and `/sub/en/sitemap` for navigation root

and

```
<VirtualHost *:80>
        ServerName test.com
[...]
        RewriteEngine on
        RewriteRule ^/(.*) http://127.0.0.1:8080/VirtualHostBase/http/test.com:80/Plone/VirtualHostRoot/$1 [L,P]
```

returns  `/sitemap` for portal and `/en/sitemap` for navigation root

and without virtual hosting, going directly to http://127.0.0.1:8080/Plone returns `/Plone/sitemap` for portal and `/Plone/en/sitemap` for navigation root. The same happen when using internal links in Link.

Reply:
do we want the complete url for "external url" ?

Reply:
I would not change how it works now, this PR just fix the bugs in `link_view` and `replace_link_variables_by_paths`. Links are already ok because the use `view.absolute_target_url()` that is the full url and remoteUrl:
https://github.com/plone/plone.app.contenttypes/blob/450888e2e4d8fa824565a83025f275473f6b1421/plone/app/contenttypes/browser/link_redirect_view.py#L100-L101

Reply:
I've updated the comment above to clarify that we're already returning the full url in the HTTP `Location` header (redirect).

Reply:
@petschki can I resolve the conversation?

Reply:
Yes